### ISSUE TITLE
Azure AD

### ISSUE THREAD

**Issue by azeem4775** [psignoret/aad-sso-wordpress#156]
Hi,

I have a strange issue which I want to clarify with you.

Plugin is setup to work with Azure AD. Wordpress general setting is set to not allow registrations and default role is set to 'Author'. in Azure AD all settings for integration i.e. client ID, secret, reply URL and Login redirect URL as well as OpenID Connect configuration endpoint etc. are correct and tested with auto-provisioning 'on'. However I want to restrict Azure AD users to a specific Azure AD Security Group members. Hence, I enabled "Azure AD group to WP role association" and set "Default WordPress role if not in Azure AD group" to Deny access. In the "WordPress role to Azure AD group map" I pasted the correct Azure AD security group object ID for 'Author' role. If I do this, the user which was auto-provisioned earlier and has the correct wordpress role and exists as member of the Azure AD group cannot login. After authentication from login.microsoftonline.com the user comes back to the Wordpress login screen with error: AAD user portal@qibaodwight.org is not a member of any group granting a role.

Please help to point out where and what I am doing wrong?

Your help will be greatly appreciated.

**Comment by psignoret** [comment]
Which version of the plugin do you currently have installed?

Did you grant the [required delegated permission](https://github.com/psignoret/aad-sso-wordpress#2-register-an-azure-active-directory-application) "Read directory data" in the Azure AD app registration?

**Comment by azeem4775** [comment]
I was using the older version - installed in August, but then read the other [post](https://github.com/psignoret/aad-sso-wordpress/issues/148) about similar issue #148  and installed the latest version right after posting this thread. Yes, I did setup the delegated permissions and checked again after reading issue #148. It didn't work but then I noticed that in permissions for Azure application there are two types. MS Azure AD and MS Graph. I update the permissions as you have instructed for both those types and it has now worked.

Thank you so much for your prompt response and help. Great work.

Regards
Muhammad
